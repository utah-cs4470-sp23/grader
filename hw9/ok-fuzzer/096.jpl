show [(- {(86.56 / 61.81), (111 + 597)}{1}), 384]

