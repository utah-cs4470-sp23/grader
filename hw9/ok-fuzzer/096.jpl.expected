global jpl_main
global _jpl_main
extern _fail_assertion
extern _jpl_alloc
extern _get_time
extern _show
extern _print
extern _print_time
extern _read_image
extern _write_image
extern _fmod
extern _sqrt
extern _exp
extern _sin
extern _cos
extern _tan
extern _asin
extern _acos
extern _atan
extern _log
extern _pow
extern _atan2
extern _to_int
extern _to_float
section .data
const0: dq 384
const1: dq 597
const2: dq 111
const3: dq 61.81
const4: dq 86.56
const5: db `(ArrayType (IntType) 1)`, 0
section .text
jpl_main:
_jpl_main:
push rbp
mov rbp, rsp
push r12
mov r12, rbp
sub rsp, 8
mov rax, [rel const0]
push rax
mov rax, [rel const1]
push rax
mov rax, [rel const2]
push rax
pop rax
pop r10
add rax, r10
push rax
mov rax, [rel const3]
push rax
mov rax, [rel const4]
push rax
movsd xmm0, [rsp]
add rsp, 8
movsd xmm1, [rsp]
add rsp, 8
divsd xmm0, xmm1
sub rsp, 8
movsd [rsp], xmm0
mov r10, [rsp - 8]
mov [rsp - 8], r10
add rsp, 8
pop rax
neg rax
push rax
mov rdi, 16
call _jpl_alloc
mov r10, [rsp + 8]
mov [rax + 8], r10
mov r10, [rsp + 0]
mov [rax + 0], r10
add rsp, 16
push rax
mov rax, 2
push rax
lea rdi, [rel const5]
lea rsi, [rsp]
call _show
add rsp, 16
add rsp, 8
add rsp, 0
pop r12
pop rbp
ret
